spec.bs

```diff
@@ -2764,11 +2766,17 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
 
 </div>
 
-<div algorithm=navigate>
-  Modify the definition of [[HTML]]'s [=navigate=] algorithm to include an extra parameter: an
-  optional [=string=] |sharedStorageContext| (default null).
+<div algorithm=navigation-must-replace-patch>
```

Comment:
Why rewrite so much of this existing algorithm? We can just modify more surgically to keep it accepting a Document, but modify the single "navigable is a fenced navigable" condition to go from Document -> [relevant global -> navigable](https://html.spec.whatwg.org/C#window-navigable) to get that navigable.

That seems less invasive right?

Reply:
I was debating on whether it was cleaner to keep the algorithm less changed, or have it set up so that it would be slightly less steps to access things. I'll keep the definition as is and do the document -> global object -> navigable traversal for the new condition.